From 5e12be4611c19a18cad78ad12e9e67d9a2cec772 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Fri, 22 Sep 2017 12:20:02 +0300
Subject: [PATCH 114/741] staging: fsl-dpaa2/mac: Request atomic context MC
 portals

commit 356f0de0359a14286b634e47317524d377d536e7 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

The MAC driver may need to issue MC commands while in atomic
context (e.g. dpaa2_mac_get_stats can be called from a critical
section), so we need to use MC portals that don't sleep while
waiting for a command response to arrive.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index 1d8f07f7e453..294511f89e51 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -490,7 +490,9 @@ static int dpaa2_mac_probe(struct fsl_mc_device *mc_dev)
 
 	dev_set_drvdata(dev, priv);
 
-	err = fsl_mc_portal_allocate(mc_dev, 0, &mc_dev->mc_io);
+	/* We may need to issue MC commands while in atomic context */
+	err = fsl_mc_portal_allocate(mc_dev, FSL_MC_IO_ATOMIC_CONTEXT_PORTAL,
+				     &mc_dev->mc_io);
 	if (err || !mc_dev->mc_io) {
 		dev_err(dev, "fsl_mc_portal_allocate error: %d\n", err);
 		err = -ENODEV;
-- 
2.17.1

